api/api/utils/moderation.py

```diff
@@ -0,0 +1,137 @@
+from typing import Literal
+
+import structlog
+
+from api.constants.moderation import DecisionAction
+from api.models.audio import (
+    Audio,
+    AudioDecision,
+    AudioDecisionThrough,
+    DeletedAudio,
+    SensitiveAudio,
+)
+from api.models.image import (
+    DeletedImage,
+    Image,
+    ImageDecision,
+    ImageDecisionThrough,
+    SensitiveImage,
+)
+from api.models.media import AbstractDeletedMedia, AbstractMedia, AbstractSensitiveMedia
+
+
+logger = structlog.get_logger(__name__)
+
+
+def perform_moderation(
+    request,
+    media_type: Literal["audio", "image"],
+    mod_objects: list[
+        type[AbstractSensitiveMedia] | type[AbstractDeletedMedia] | type[AbstractMedia]
+    ],
+    action: DecisionAction,
+):
+    """
+    Perform bulk moderation on the given models.
+
+    If the decision action is forward, ``mod_objects`` is a queryset of
+    ``Media`` items. We can get the UUIDs from the ``identifier`` field.
+
+    If the decision action is reverse, ``mod_objects`` is a queryset of
+    ``SensitiveMedia`` or ``DeletedMedia`` items. We can get the UUIDs
+    from the ``media_obj_id`` field.
+
+    Note that bulk moderation will not resolve any open reports. It is
+    up to the moderator to manually link open reports with the
+    appropriate decisions and resolve them.
```

Comment:
Yes, this is a bit of manual work that can be automated if we auto-close reports. However, there are two main issues preventing that.

> The main case we need to _avoid_ that I can think of is if we marked an image as sensitive based on one report, but _another_ report (that gets closed for having the same reason) actually contains information or context which would lead us to _deindex_ the image.
> &mdash; @zackkrida

> Also need to keep in mind reversing a bulk action — if creating one results in reports being resolved, reversing it ought to reopen them
> &mdash; @stacimc 

With this limitation in mind, I decided to keep the reports unchanged for now and potentially revisit this in the future.

Reply:
Ideally, with moderation happening regularly, the need to triage multiple concurrent reports of a single work will be very infrequent.

Reply:
Does that mean we should implement report closing functionality as a part of bulk moderation? It can be pretty complex in terms of logic so I would ideally like that to be a different issue-PR pair to avoid complicating this PR further.